### ISSUE TITLE
🎛️ Migrate Skip Link to a Stimulus Controller

### ISSUE THREAD

**Issue by lb-** [wagtail/wagtail#9810]
> ℹ️ **Part of the [Stimulus  🎛️ RFC 78](https://github.com/wagtail/rfcs/pull/78)**

### Is your proposal related to a problem?

The Skip Link widget, which is used for keyboard / screen reader control allowing the user to skip to the `main` element, can be migrated to the new Stimulus controller convention.

<img width="1810" alt="Screen Shot 2022-12-20 at 8 16 11 am" src="https://user-images.githubusercontent.com/1396140/208537249-308033f6-2857-4573-8e87-f5d59ab5c004.png">

### Describe the solution you'd like

* Create a stimulus controller `w-skip-link` that will replace the existing ad-hoc JS implementation
* The behaviour should be exactly the same as current state
* Controller should be written in TypeScript
* Ensure that the existing unit tests are updated to reflect the new DOM usage
* This does not need a Storybook story
* No upgrade considerations in the documentation should be needed for this change

#### POC Implementation

A rough POC has been done of this and can be found at the following commits/files. Be sure to consider any recent changes to to the actual Stimulus code that has merged in.

* https://github.com/lb-/wagtail/blob/rfcs/stimulus-ui/client/src/controllers/SkipLinkController.ts
* https://github.com/lb-/wagtail/blob/rfcs/stimulus-ui/client/src/controllers/SkipLinkController.test.js
* https://github.com/lb-/wagtail/blob/rfcs/stimulus-ui/wagtail/admin/templates/wagtailadmin/skeleton.html#L78

#### Current files

* [wagtail/admin/templates/wagtailadmin/skeleton.html](https://github.com/wagtail/wagtail/blob/main/wagtail/admin/templates/wagtailadmin/skeleton.html)
* [client/src/includes/initSkipLink.ts](https://github.com/wagtail/wagtail/blob/main/client/src/includes/initSkipLink.ts)
* [client/src/includes/initSkipLink.test.js](https://github.com/wagtail/wagtail/blob/main/client/src/includes/initSkipLink.test.js)
* [client/src/entrypoints/admin/wagtailadmin.js](https://github.com/wagtail/wagtail/blob/main/client/src/entrypoints/admin/wagtailadmin.js#L44) - this import can calling usage will no longer be needed
